tensorboard/components/tf_data_selector/storage-utils.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+namespace tf_data_selector {
+
+export function decodeIdArray(str: string): Array<number> {
+  return str.split(',').map(idStr => parseInt(idStr, 36)).filter(Boolean);
```

Comment:
I fully support de-generalizing the the over-generalized `encodeURIComponent(btoa(JSON.stringify({})))` in tf_storage/storage.ts. However base36 might make URIs longer than base64.

How about a well-defined one-off encoding, for each and every individual parameter? For example:

```js
> f = (a) => a.map(encodeURIComponent).join(':')
> '#tags=' + f(['normal', 'person'])
'#tags=normal:person'
> '#tags=' + f(['i&am=evil', 'lol#lol'])
'#tags=i%26am%3Devil:lol%23lol'
```

This gives us *really nice* looking URIs in the average case, and the only get uglier as the user gets more creative with naming. Also, encodeURIComponent() escapes colon, since it's much more conservative than the RFC: https://tools.ietf.org/html/rfc3986 i.e. `pchar = unreserved / pct-encoded / sub-delims / ":" / "@"`.

Reply:
Oh wow! It turns out base36 encoding in JS could be more accurately described as similar to CEscape:

```
> 'run-name'.toString(36)
'run-name'
> 'run-\0name'.toString(36)
'run-\u0000name'
```

So this should make the URI#fragment content short and readable. Bravo.